torch/_dynamo/variables/builder.py

```diff
@@ -1396,6 +1397,10 @@ def wrap_tensor(self, value: torch.Tensor):
         # and recursively install corresponding guard for each inner attribute.
         if is_traceable_wrapper_subclass(value):
             self.install_guards(GuardBuilder.TYPE_MATCH)
+            install_guard(
+                SubclassAttrListSource(source).make_guard(GuardBuilder.EQUALS_MATCH)
```

Comment:
cc @tugsbayasgalan - we'll need to fix this for nested subclasses too. Dynamo generate a type match (and also am match on tensor_flatten attributes), and we should generate these guards recursively for inner tensors. (We can fix this independently of the metadata guard [issue](https://github.com/pytorch/pytorch/issues/114405))